### DISCUSSION TITLE
how to use custom yolov8 model for object detection using cpu

### DISCUSSION THREAD

**Discussion by Aadi4401** [blakeblackshear/frigate#15568] [Ask A Question]
i have a yolov8 custom model for ppe detection i am using it in config.yaml file 

this is config.yaml file code

mqtt:
  enabled: false

cameras:
  school1: # <------ Name the camera
    enabled: true
    ffmpeg:
      inputs:
        - path: rtsp:myrtsp
          roles:
            - detect
            - record
    detect:
      enabled: true # <---- disable detection until you have a working camera feed
      width: 1280
      height: 720

    zones:
      lobby:
        coordinates: 
          0.376,0.096,0.612,0,0.626,0.427,0.674,0.753,0.362,0.863,0.379,0.595,0.377,0.342,0.372,0.155
        inertia: 3
        loitering_time: 0
        objects:
          - person
          - NO-Hardhat
          - NO-Safety Vest
    record:
      enabled: true
      events:  # Event-based recording
        retain:
          default: 7  # Retain recordings for 7 days
        pre_capture: 5  # Record 5 seconds before the event
        post_capture: 5 # Record 5 seconds after the event ends

      # retain:
      #   days: 7  # Retain recordings for 7 days
    review:
      alerts:
        required_zones: lobby
      detections:
        required_zones: lobby
    motion:
      mask: 
        0.176,0.274,0.447,0.027,0.593,0.043,0.758,0.192,0.855,0.313,0.991,0.53,0.924,0.861,0.88,0.959,0.142,0.973,0.082,0.366
detectors:
  custom:
    type: cpu
    model:
      path: /config/custom_models/ppe3_float32.tflite

  school2:
    enabled: true
    ffmpeg:
      inputs:
        - path: rtsp:myrtsp
          roles:
            - detect
    detect:
      enabled: false # <---- disable detection until you have a working camera feed
      width: 1280
      height: 720
  school3:
    enabled: true
    ffmpeg:
      inputs:
        - path: rtsp:myrtsp
          roles:
            - detect
    detect:
      enabled: false
      width: 1280
      height: 720
  school4:
    enabled: true
    ffmpeg:
      inputs:
        - path: rtsp:myrtsp
          roles:
            - detect
    detect:
      enabled: false
      width: 1280
      height: 720

version: 0.14

this is docker-compose code
# version: "3"
services:
  devcontainer:
    container_name: frigate-devcontainer
    # add groups from host for render, plugdev, video
    group_add:
      - "109" # render
      - "110" # render
      - "44"  # video
      - "46"  # plugdev
    shm_size: "256mb"
    build:
      context: .
      dockerfile: docker/main/Dockerfile
      # Use target devcontainer-trt for TensorRT dev
      target: devcontainer
    ## Uncomment this block for nvidia gpu support
    # deploy:
    #       resources:
    #           reservations:
    #               devices:
    #                   - driver: nvidia
    #                     count: 1
    #                     capabilities: [gpu]
    environment:
      YOLO_MODELS: yolov7-320
    devices:
      - /dev/bus/usb:/dev/bus/usb
      # - /dev/dri:/dev/dri # for intel hwaccel, needs to be updated for your hardware
    volumes:
      - .:/workspace/frigate:cached
      - ./web/dist:/opt/frigate/web:cached
      - /etc/localtime:/etc/localtime:ro
      - ./config:/config
      - ./debug:/media/frigate
      - /dev/bus/usb:/dev/bus/usb
    ports:
      - "5000:5000"  # Expose port 5000 for Frigate web UI and API
      - "8554:8554"  # RTSP stream port
      - "8555:8555/udp"  # WebRTC signaling port

  mqtt:
    container_name: mqtt
    image: eclipse-mosquitto:1.6
    ports:
      - "1883:1883"

someone please provide step by step guidance for using custom yolov8 model in frigate
when ever i use this code for detection frigate is stopped when i comment this part it is working 
detectors:
  custom:
    type: cpu
    model:
      path: /config/custom_models/ppe3_float32.tflite 

these are the classes in my model 
{0: 'Hardhat', 1: 'Mask', 2: 'NO-Hardhat', 3: 'NO-Mask', 4: 'NO-Safety Vest', 5: 'Person', 6: 'Safety Cone', 7: 'Safety Vest', 8: 'machinery', 9: 'vehicle'}

**Comment by NickM-27** [comment]
Only mobiledet & mobilenet models are supported on the coral

**Reply by Aadi4401** [reply]
i've already converted my yolov8 ppe3 model to tflite model

**Reply by NickM-27** [reply]
right, but like I said yolov8 is not supported, only mobiledet and mobilenet are. Yolov8 was supported for a short while before it had to be removed, you can see https://github.com/blakeblackshear/frigate/pull/10717 for more info
